Sources/performance-worker/WorkerService.swift

```diff
@@ -41,13 +41,42 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
       self.role = role
     }
 
-    init(server: GRPCServer) {
-      self.role = .server(server)
+    init(server: GRPCServer) async throws {
+      let initialState = try await ServerStats()
+      self.role = .server(server, initialState)
+
     }
 
     init(client: GRPCClient) {
       self.role = .client(client)
     }
+
+    public func getServer() -> GRPCServer? {
+      switch self.role {
+      case let .server(server, _):
+        return server
+      default:
+        return nil
+      }
+    }
+
+    public func getInitialServerStats() -> ServerStats? {
+      switch self.role {
+      case let .server(_, initialServerStats):
+        return initialServerStats
+      default:
+        return nil
+      }
+    }
+
+    public mutating func setInitialServerStats(newStats: ServerStats) {
```

Comment:
I wouldn't bother with the label here, the name of the function already indicates what the function does and what the expected parameter is. 

```suggestion
    public mutating func setInitialServerStats(_ newStats: ServerStats) {
```